### PR TITLE
Save gallery

### PR DESCRIPTION
**Pull request by demoran23**
I was looking into why I couldn't save my gallery using the latest code.  It appears to be related to missing files.  It looks like we don't have id based chunks anymore, and since they would throw the app would just sit there without doing anything.

I cleaned up the code a bit, but there is still an outstanding issue.  All of the specified files are making their way into the zip, but we're seeing some errors in the console.

I also had some trouble with reformatting code, since my editor is configured differently from yours.  It would be helpful if you'd include an .editorconfig file that my editor could obey, to prevent format changes in a large file when I format my code.

### CODE REVIEW

File: src/gallery/store.js [hunk 1]
```diff
@@ -97,22 +97,22 @@ export default new Vuex.Store({
 
     buildStandaloneData: function(state, inputArray) {
       state.library = state.library || {};
-      _.each(inputArray, function( item ) { 
-        state.library.[item.key] = Object.freeze( item.value ); 
+      _.each(inputArray, function( item ) {
+        state.library[item.key] = Object.freeze( item.value );
```
Comment (demoran23) [inline src/gallery/store.js:101]: This is the actual fix.  Sorry, I didn't realise there were so many whitespace changes.

Comment (joonaspaakko) [reply src/gallery/store.js:101]: I'm pretty sure this a non issue related to how I don't really care much about committing fully functional code because it's all experimental until a released build (I'll explain that)... **or** the issue could also be related to you testing it using the `watch:dev` command. So I think they're both basically causing the same issue. Jszip doesn't like it when you try to serve it files that do not exist and this causes the saving process to stop. 

Now there are 3 chunk files that I've not been able to give a name to and their `contenthash` in the filename keeps changing every now and then so probably in quite a few commits I've not updated the `save-gallery.vue` → `files` array and it's trying to add files to jszip that don't exist. 

Whenever I want to test saving the standalone gallery I usually do a `build`, then update the `save-gallery.vue` → `files` array using the exact files from the `dist/gallery/chunks` folder and build again. But like I said, I only do this if I'm specifically testing the standalone gallery saving or something in the standalone gallery, so there are likely a kazillion commits where the saving process just doesn't work in the `dist-zip`.

The other thing is that if you're serving the files using `watch:dev`, it gives you a slightly different set of chunk files and so obviously the `save-gallery.vue` → `files` array is not going to match at all. I've taken to only testing the standalone gallery in a complete `build` to give it a fair chance. So I'd recommend that.

I'm fairly confident the issue lies somewhere in there given the changes you made, and on that note:

`.[` is definitely a syntax error, but I was surprised it had not caused any issues so far in my tests. So I looked at my test gallery [here](https://joonaspaakko.github.io/ale-test-new/gallery.1632088147952.js) (you can prettify the code in Chrome's dev tools sources tab... and search for `buildStandaloneData:`) and I noticed that it's actually fixed there in my built code... perhaps when the code is minified? I'm not sure, but what I do know is that when I use the commands `watch:dev` or `build`, somewhere during these processes the issue gets fixed (in my end anyways).

> The thing I can't figure out is why this would be any different on your end...

image

But then there's the second thing, which is the standalone gallery saving process. Whether or not this `.[`is an actual issue in the built code, the thing is, it's not related to the saving process. It's related to opening the gallery pages in the standalone gallery.

And if I'm reading the changes that you made in the `save-gallery.vue` file correctly, it's suppressing errors... and while that might help you get through the saving process, I'm not overly confident that it would result in fully functional standalone gallery.

-----

**So in conclusion:**

1. I think the issue has to do with quirks in the dev environment.
2. The `.[` syntax errors, which I found a few more of don't seem to be an issue, but there's still no reason not to fix them.
3. I don't think suppressing jszip errors is a good approach or necessary.

P.S. I think your editor might have the `trimTrailingWhitespace` setting on, which is causing the  whitespace changes. I don't like how that setting doesn't just remove the whitespace at the end of a line, but it also removes indentation on empty lines. When the code comes out minified anyways, I don't much care how much whitespace exists in the source files so I keep it disabled. I will be adding `.vscode/settings.json` → `"files.trimTrailingWhitespace": false`.

Comment (demoran23) [reply src/gallery/store.js:101]: You're right, I was testing with `watch:dev`!  That makes sense.  

In my testing, I found that you could pass a function to `chunkFilename`; I bet we could use that.  `[name]` falls back to `[id]`, but maybe we could be like "Well, if you don't have a name, use this known value instead".

With respect to the syntax error, I just happened to see it highlighted in my editor and was surprised that webpack didn't blow up in the first place.

Speaking of blowing up in the first place, you're right about blowing up instead of silently failing / doing part of the job.

With respect to editors, I use IntelliJ IDEA for javascript (and actually use Rider for .net stuff now too).

File: webpack.config.js [hunk 2]
```diff
@@ -10,6 +10,15 @@ const { CleanWebpackPlugin } = require('clean-webpack-plugin');
 const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
 const FileManagerPlugin = require('filemanager-webpack-plugin-fixed');
 
+let runningChunkId = 0;
+const runningChunkMap = {};
+
+const getSafeChunkFilename = (extension) => (a) => {
```
Comment (demoran23) [inline webpack.config.js:16]: Ok, so here's the happy balance between being resilient and being sloppy.  Starting with an id of 1, we incrementally name each unknown chunk.

Then, for chunks 1 to 10 (we only have 2 unknown now), we try to add them to the exported gallery (cf line 329).  This means that the code should never need to be updated again, but you'll see a bit of red as it attempts to get chunks 3 through 8 as it does its gallery export.

Comment (joonaspaakko) [reply webpack.config.js:16]: Webpack config in general is not my strong suit, but I looked into this again not what you brought it up and I think I may have found the solution I wanted. I approached it from a different angle this time. Every named chunk had a name because I had given them names during import, so I looked through the imports if there was anything to improve, but there was still those few that I just couldn't name without causing other issues...
